srml/metadata/src/lib.rs

```diff
@@ -286,6 +286,18 @@ pub enum StorageEntryModifier {
 	Default,
 }
 
+/// All metadata of the storage.
+#[derive(Clone, PartialEq, Eq, Encode)]
+#[cfg_attr(feature = "std", derive(Decode, Debug, Serialize))]
+pub struct StorageMetadata {
+	/// The common prefix used by all storage entries.
+	pub prefix: DecodeDifferent<&'static str, StringBuf>,
+	/// The instance identifier. This identifier is appended to each storage entry prefix to make it
+	/// unique per instance.
+	pub instance: Option<DecodeDifferent<&'static str, StringBuf>>,
```

Comment:
should we just add this into the prefix ? I think both are fine actually

Reply:
That will not work. `prefix` is for example `Staking`.
But the final prefix for a storage entry will be calculated like:
`PREFIX_FOR_Staking_{ValueName}{Instance}`

Reply:
yes what I mean is that prefix should be the final one.

Reply:
an instantiated module shouldn't differ in any way to a regular module from metadata point of view

Reply:
I don't wanted to break storage, but yeah. You are right. Better do it now than later.

Reply:
(Removed)